src/openrct2/ride/Vehicle.cpp

```diff
@@ -7419,15 +7412,25 @@ void Vehicle::UpdateAdditionalAnimation()
                     else
                         seat_rotation++;
 
-                    animation_frame = (seat_rotation - 4) & 7;
-                    Invalidate();
+                    int8_t targetSeatRotation = (seat_rotation - 4);
+                    if (targetSeatRotation != animation_frame)
+                    {
+                        if (targetSeatRotation >= 0)
+                            animation_frame = targetSeatRotation;
+                        else
+                            // % is remainder function in C. This requires some reworking to use as a modulus for
+                            // negative numbers
+                            animation_frame = (vehicleEntry->AnimationFrames - 1)
+                                - ((-targetSeatRotation - 1) % vehicleEntry->AnimationFrames);
```

Comment:
The module operator works fine for negative integers too. You may be having troubles because this seems to be working with offsets and masks before. This could use some rewriting to make the code understandable.

Reply:
Interpreting `-1 % 8` was returning 255 and failing replays. This does not fail replays. Dunno why.

Reply:
I found a full explanation of why the modulus operator doesn't work, although not why `-1 % 8` returns 255.

C integer division rounds toward zero.
C also defines the modulus operator as `a = a / b + a % b`

Since a = -1 and b = 8, the equation becomes:
`-1 = (-1) / 8 + (-1) % 8`
Integer division is easy to solve, it's just zero. Therefore:
`-1 = (-1) % 8`

Reply:
@Broxzier

Reply:
Rewrite this section completely its a mess
do something like the following there is no need to modulo anything as we already have an if:
```cpp
if (seat_rotation - 4 < 0){
   animation_frame = seat_rotation + (vehicleEntry->AnimationFrames - 1) - 4;
}else{
    animation_frame = seat_rotation - 4;
}
Invalidate();
```